### DISCUSSION TITLE
Pages build and deployment Error

### DISCUSSION THREAD

**Discussion by green-study** [community/community#60751] [Actions] [answered]
### Select Topic Area

Question

### Body

![image](https://github.com/community/community/assets/70996958/5eec42fe-24cc-4a7f-9e5a-c7187dd5b7eb)

There was no problem before, but suddenly it doesn't deploy.
And there was no problem when the source code was built locally, and there was no merge error.
May I know the solution?

**Comment by JamesMGreene** [comment, marked as answer]
**TL;DR:** GitHub Pages only allows one concurrent deployment per branch.

Assuming you are attempting to deploy to Pages using a custom Actions workflow, the error message mentioned that the `Deployment request failed... due to in progress deployment`. This suggests you had multiple executions of this workflow running concurrently.

In your Actions workflow definition, try adding a [`concurrency` configuration](https://docs.github.com/en/actions/using-jobs/using-concurrency) something like the following (taken from the [Pages starter workflow template for Hugo](https://github.com/actions/starter-workflows/blob/2bb20df369386c3e20630fce032d3ab4915373a1/pages/hugo.yml#L18-L22)):

```yaml
# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false
```
